webui/react/src/hooks/useModal/useModalHyperparameterSearch.tsx

```diff
@@ -0,0 +1,770 @@
+import { SelectValue } from 'antd/lib/select';
+import yaml from 'js-yaml';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import Link from 'components/Link';
+import SelectFilter from 'components/SelectFilter';
+import { useStore } from 'contexts/Store';
+import { maxPoolSlotCapacity } from 'pages/Clusters/ClustersOverview';
+import { paths } from 'routes/utils';
+import { createExperiment } from 'services/api';
+import { V1MetricNamesResponse } from 'services/api-ts-sdk';
+import { detApi } from 'services/apiConfig';
+import { readStream } from 'services/utils';
+import { Primitive } from 'shared/types';
+import { clone, flattenObject, isBoolean, unflattenObject } from 'shared/utils/data';
+import { DetError, isDetError } from 'shared/utils/error';
+import { roundToPrecision } from 'shared/utils/number';
+import { routeToReactUrl } from 'shared/utils/routes';
+import { validateLength } from 'shared/utils/string';
+import { ExperimentBase, ExperimentSearcherName, Hyperparameter,
+  HyperparameterType, ResourcePool, TrialDetails, TrialHyperparameters, TrialItem } from 'types';
+import { alphaNumericSorter } from 'utils/sort';
+
+import useModal, { ModalHooks as Hooks, ModalCloseReason } from './useModal';
+import css from './useModalHyperparameterSearch.module.scss';
+
+interface Props {
+  experiment: ExperimentBase;
+  trial?: TrialDetails | TrialItem;
+}
+
+export interface ShowModalProps {
+  initialModalProps?: ModalFuncProps;
+  trial?: TrialDetails | TrialItem;
+}
+
+interface ModalHooks extends Omit<Hooks, 'modalOpen'> {
+  modalOpen: (props?: ShowModalProps) => void;
+}
+
+interface SearchMethod {
+  description: string;
+  displayName: string;
+  name: `${ExperimentSearcherName}`;
+}
+
+const SearchMethods: Record<string, SearchMethod> = {
+  ASHA: {
+    description: `Automated HP search multi-trial experiment that will stop poor 
+  performing trials early as it searches the HP space.`,
+    displayName: 'Adaptive ASHA',
+    name: 'adaptive_asha',
+  },
+  AsyncHalving: {
+    description: `Automated HP search multi-trial experiment that will stop poor
+  performing trials early as it searches the HP space.`,
+    displayName: 'Async Halving',
+    name: 'async_halving',
+  },
+  Grid: {
+    description: `Brute force evaluates all possible hyperparameter configurations 
+  and returns the best.`,
+    displayName: 'Grid',
+    name: 'grid',
+  },
+  Random: {
+    description: `Evaluates a set of hyperparameter configurations chosen at 
+  random and returns the best.`,
+    displayName: 'Random',
+    name: 'random',
+  },
+} as const;
+
+interface HyperparameterRowValues {
+  count?: number;
+  max?: number,
+  min?:number
+  type: HyperparameterType,
+  value?: number | string,
+}
+
+const useModalHyperparameterSearch = ({ experiment, trial: trialIn }: Props): ModalHooks => {
+  const { modalClose, modalOpen: openOrUpdate, modalRef, contextHolder } = useModal();
```

Comment:
**nit suggestion:** since `contextHolder` isn't actually used
```
const { modalClose, modalOpen: openOrUpdate, ...modalProps } = useModal();

...

return { modalClose, modalOpen, modalRef, ...modalProps };
```